File: runme/mcgen/local/datan/datsrc/minglp.f
      DOUBLE PRECISION FUNCTION MINGLP(X,N)
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      COMMON /CMINGH/ T(100),HIST(100),DELTAT,NT,NEVENT
      DIMENSION X(N)
      PARAMETER(ZERO=0.D0)
      FNORM=DBLE(NEVENT)*DELTAT
      MINGLP=ZERO
      DO 10 I=1,NT
C GI is the value of the probability density of the population
C at T(I) (by replacing the RHS of the following statement it can
C be changed from normal to any desired distribution)
        GI=SDNORM(T(I),X(1),X(2))
C normalize to number of events in sample
        GI=FNORM*GI
        ALAM=GI
        IF(ALAM.GT.ZERO) THEN
          NI=NINT(HIST(I))
          ALNLAM=LOG(ALAM)
          ALNNI=GLNGAM(DBLE(NI+1))
          MINGLP=MINGLP+ALNNI-HIST(I)*ALNLAM+ALAM
        END IF
   10 CONTINUE
      END

File: runme/mcgen/local/datan/datsrc/minglp.c
#include <math.h>
#include "datsrc.h"

/* Common Block Declarations */

#if defined(__TURBOC__)
extern
#endif
struct cmingh_ {
    double t[100], hist[100], deltat;
    integer nt, nevent;
} LIBDATA cmingh_;

double LIBFUNC minglp_(double const P_T *x, integer const P_T *n)
{
    /* System generated locals */
    integer i__1;
    double ret_val, d__1;

    /* Local variables */
    integer i, ni;
    double alam, alnlam, alnni, fnorm, gi;

    /* Parameter adjustments */
    --x;

    /* Function Body */
    fnorm = (double) cmingh_.nevent * cmingh_.deltat;
    ret_val = 0.;
    i__1 = cmingh_.nt;
    for (i = 1; i <= i__1; ++i) {
/* GI is the value of the probability density of the population */
/* at T(I) (by replacing the RHS of the following statement it can */
/* be changed from normal to any desired distribution) */
        gi = sdnorm_(&cmingh_.t[i - 1], &x[1], &x[2]);
/* normalize to number of events in sample */
        gi = fnorm * gi;
        alam = gi;
        if (alam > 0.) {
            ni = nint(cmingh_.hist[i - 1]);
            alnlam = log(alam);
            d__1 = (double) (ni + 1);
            alnni = glngam_(&d__1);
            ret_val = ret_val + alnni - cmingh_.hist[i - 1] * alnlam + alam;
        }
    }
    return ret_val;
} /* minglp_ */

